src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -141,19 +141,19 @@ public bool CanNavigateToPosition(Workspace workspace, DocumentId documentId, in
             return CanMapFromSecondaryBufferToPrimaryBuffer(workspace, documentId, vsTextSpan);
         }
 
-        public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options, CancellationToken cancellationToken)
+        public bool TryNavigateToSpan(Workspace workspace, DocumentId documentId, TextSpan textSpan, OptionSet options, bool allowInvalidSpan, CancellationToken cancellationToken)
         {
             return TryNavigateToLocation(workspace,
                 documentId,
                 _ => textSpan,
-                text => GetVsTextSpan(text, textSpan),
+                text => GetVsTextSpan(text, textSpan, allowInvalidSpan),
                 options,
                 cancellationToken);
 
-            static VsTextSpan GetVsTextSpan(SourceText text, TextSpan textSpan)
+            static VsTextSpan GetVsTextSpan(SourceText text, TextSpan textSpan, bool allowInvalidSpan)
             {
                 var boundedTextSpan = GetSpanWithinDocumentBounds(textSpan, text.Length);
-                if (boundedTextSpan != textSpan)
+                if (boundedTextSpan != textSpan && !allowInvalidSpan)
```

Comment:
we're *always* resilient to the span being requested not being in the span of the tree. however, by default, we will log a watson for that case right below this.  Now, in the case of showing stale navto entries, we do not report such a watson as being not within bounds of hte doc is not considered an exceptional circumstance.